### PR TITLE
asm: loongarch64: Implementation of clobber_abi

### PR DESCRIPTION
**Pull request by heiher**
r? @Amanieu

### CODE REVIEW

File: compiler/rustc_target/src/asm/mod.rs [hunk 1]
```diff
@@ -1022,6 +1027,21 @@ impl InlineAsmClobberAbi {
...
+                    r4, r5, r6, r7, r8, r9, r10, r11,
+                    // t0-t8
+                    r12, r13, r14, r15, r16, r17, r18, r19, r20,
+                    // fa0-fa7
+                    f0, f1, f2, f3, f4, f5, f6, f7,
+                    // ft0-ft15
+                    f8, f9, f10, f11, f12, f13, f14, f15,
+                    f16, f17, f18, f19, f20, f21, f22, f23,
+                }
+            },
```
Comment (heiher) [inline compiler/rustc_target/src/asm/mod.rs:1044]: LoongArch Register Convention: https://loongson.github.io/LoongArch-Documentation/LoongArch-ELF-ABI-EN.html

File: compiler/rustc_target/src/asm/mod.rs [hunk 2]
```diff
@@ -880,6 +881,10 @@ impl InlineAsmClobberAbi {
                 "C" | "system" | "efiapi" => Ok(InlineAsmClobberAbi::RiscV),
                 _ => Err(&["C", "system", "efiapi"]),
             },
+            InlineAsmArch::LoongArch64 => match name {
+                "C" | "system" | "efiapi" => Ok(InlineAsmClobberAbi::LoongArch),
```
Comment (Amanieu) [inline compiler/rustc_target/src/asm/mod.rs:885]: I just noticed this, but this should not include `efiapi`, which is not a valid ABI on loongarch.

Comment (heiher) [reply compiler/rustc_target/src/asm/mod.rs:885]: Thanks for catching that mistake. I'll make the necessary changes.

**Pull request merged**